runtime/src/main/java/org/corfudb/runtime/collections/CorfuStore.java

```diff
@@ -71,6 +73,27 @@ private Timestamp getTimestamp() {
                 .build();
     }
 
+    @Nonnull
+    public <K extends Message, V extends Message, M extends Message>
+    Table<K, V, M> openTable(@Nonnull FullyQualifiedTableName fqTableName,
```

Comment:
From the client's usage perspective isn't this adding the number of steps ?
Previously they just called `openTable(namespace, tablename, key, value, metadata, TableOptions)`
Now with this proposed change their openTable will be split into 3 invocations:
1. FullQualifiedTableName creation
2. TableDescriptor creation
3. openTable() with the 2 parameters above
How is splitting 1 call into 3 separate calls going to simplify their usage?

Reply:
It expected that FullQualifiedTableName will be created once during the operation of table.open() and then will be reused everywhere in underlying layers, which make code more coherent and more navigable with an IDE and makes it more transparent for understanding, in compare to many parameters copy-pasted and saved independently in multiple classes (I mean kClas, vClass, mClass - you can see they spread across the entire system and going together all the time).

Also, I can't just provide a new api and get rid of the old one, because it's public api. At the same time if we don't start changing it, we will stuck forever with a cumbersome old api, we have to do something